src/geometry/s2_cell_union.rs

```diff
@@ -0,0 +1,33 @@
+//! A cell union, re-exported from the s2 crate.
+pub use s2::cellunion::CellUnion;
+
+use crate::geometry::Aabb;
+use crate::math::base::PointCulling;
+use crate::math::sat::ConvexPolyhedron;
+use crate::math::FromPoint3;
+use nalgebra::{Point3, RealField};
+use s2::{cell::Cell, cellid::CellID, region::Region};
+
+impl<S> PointCulling<S> for CellUnion
+where
+    S: RealField,
+    f64: From<S>,
+{
+    fn contains(&self, p: &Point3<S>) -> bool {
+        self.contains_cellid(&CellID::from_point(p))
+    }
+
+    fn intersects_aabb(&self, aabb: &Aabb<S>) -> bool {
```

Comment:
maybe here it is worth to mention that the intersection is carried on the rect bound of the cell union